railties/lib/rails/generators/rails/app/templates/config/initializers/content_security_policy.rb.tt

```diff
@@ -17,7 +17,8 @@
 #   end
 #
 #   # Generate session nonces for permitted importmap, inline scripts, and inline styles.
-#   config.content_security_policy_nonce_generator = ->(request) { request.session.id.to_s }
+#   # For more on what value to use: https://guides.rubyonrails.org/security.html#adding-a-nonce
+#   config.content_security_policy_nonce_generator = ->(request) { request.session.id || SecureRandom.hex(16) }
```

Comment:
I don't think this will solve the problem entirely. Turbo is going to be expecting the nonce to be the same on the next request and if the first one is randomly generated and the next one is the session.id, these will never match.

The approach I have used that works is: 

```
config.content_security_policy_nonce_generator = ->(request) { request.session[:nonce] ||= SecureRandom.hex }
```

Basically, if you are going to use session-based nonces, then just go ahead and both create a session and store the nonce value there.  There may be a very modest improvement in security by not including the session id in the page source as well.